### DISCUSSION TITLE
[Tutorial] Sectors Extension

### DISCUSSION THREAD

**Discussion by William8Work** [filecoin-project/lotus#6449] [Tutorials]
For v1_1 sectors, including CC and deal sectors, can be extended for additional period of life before expiration. Here are some of the steps you can do and several comments around the sector extension:

**1. Extend an existing sector**

`lotus-miner sectors extend`

**2. v1 sectors** (created during space race time) can be extended to up to 540 days.

**3. v1_1 proof sectors (since mid December?)** can be extended for 540 days each time when running the extend command above, up to 5 years in total life.

**4. Sector Expiry**
When a sector expires (without extension or upgrade), the pledge amount will be returned to your balance right away. There will be zero power remaining in the sector for reward calculation. And you need to manually remove the sector `lotus-miner sector remove` command.

**5. Sector Upgrade**
When a CC sector is upgraded, see [another tutorial here](https://github.com/filecoin-project/lotus/discussions/5987), the life span and pledge will change into a new sector. The original CC sector needs to be removed manually.

**6. Sector Extension**

- If no extension or upgrade, when a deal sector is expired, the sector becomes inactive, and its pledge amount will be return to your balance.
- If a CC sector is extended, no FIL is required to be added, no pledge will be returned. The sector is simply updated with its existing pledge and new expiration date.
- When a verified-deal sector is extended, the sector power will be adjusted. The 10x power from the original verified deal will be updated according to this [https://spec.filecoin.io/#section-systems.filecoin_mining.sector.sector-quality](https://spec.filecoin.io/#section-systems.filecoin_mining.sector.sector-quality)
-

**Comment by jennijuju** [comment]
Thanks for writing this up! One small bit, It’s actually v1_1 proof not v2!

**Reply by jennijuju** [reply]
And I think the last point is a bit miss leading. The verified deal doesn’t become a regular deal and it still keeps the adjusted power. However, instead of 10x, the quality adjusted power will be different.          
<img width="661" alt="image" src="https://user-images.githubusercontent.com/42981373/121601339-d021d680-ca13-11eb-87af-428804dd2eda.png">

See: https://spec.filecoin.io/#section-systems.filecoin_mining.sector.sector-quality

**Reply by s0nik42** [reply]
The formula you shared applied since the sector is created.so if a verified deal is expired in a sector its contribution to the sector is x1. When you extend a sector it doesn't extend the deals inside indeed.

Any mechanism planned to extend deals?

**Reply by jennijuju** [reply]
It doesn’t extend the deal but the adjusted power is recalculated based on new  deal weight upon extension.
Deal renewal fip is more than welcomed if doesn’t exist already!
